functional_tests/functional/functional_test.go

```diff
@@ -61,6 +61,7 @@ const (
 	manifestsDir                           = "manifests"
 	eksValuesDir                           = "expected_eks_values"
 	kindValuesDir                          = "expected_kind_values"
+	helmUninstallTimeout                   = 5 * time.Minute
```

Comment:
I have no concerns adding this. But I think the default is 5mins already (at least when invoked from cli).

If you want it to wait for pods to be fully removed, try the DeletionPropagation option set to foreground instead.

Reply:
I see. The problem with `Wait` https://github.com/helm/helm/blob/99c065789ef8c45bade24d4bc2d33432595de956/pkg/action/uninstall.go#L133-L139. We need to delete it or apply timeout. Otherwise, it's useless and confusing. Maybe we can rely on k8s client, but I've seen a lot of errors like [this](https://github.com/signalfx/splunk-otel-collector-chart/actions/runs/13734405876/job/38416507267?pr=1704). So maybe we should set more than 5 minutes. BTW I've submitted a separate PR for this https://github.com/signalfx/splunk-otel-collector-chart/pull/1706

Reply:
Ok, the error that I mentioned is because of a bug in [this line](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/.github/workflows/functional_test_v2.yaml#L186). `helm list | grep -q "^sock$"` is always false because `helm list` output is a table. The upgrade tests are pretty broken

Reply:
I removed the timeout for now and solved the failures by moving installation of the initial (pre-upgrade) release in the go tests so the lease is acquired prior to that

Reply:
@jinja2 looks like the timeout is actually needed. [This failure](https://github.com/signalfx/splunk-otel-collector-chart/actions/runs/13796823642/job/38590881913?pr=1704) sounds like particularly caused by this